pkg/controllers/providers/instancetype/capacity/controller.go

```diff
@@ -38,18 +39,23 @@ import (
 
 type Controller struct {
 	kubeClient           client.Client
+	cloudProvider        cloudprovider.CloudProvider
 	instancetypeProvider *instancetype.DefaultProvider
 }
 
-func NewController(kubeClient client.Client, instancetypeProvider *instancetype.DefaultProvider) *Controller {
+func NewController(kubeClient client.Client, cloudProvider cloudprovider.CloudProvider, instancetypeProvider *instancetype.DefaultProvider) *Controller {
 	return &Controller{
 		kubeClient:           kubeClient,
+		cloudProvider:        cloudProvider,
 		instancetypeProvider: instancetypeProvider,
 	}
 }
 
 func (c *Controller) Reconcile(ctx context.Context, node *corev1.Node) (reconcile.Result, error) {
 	ctx = injection.WithControllerName(ctx, "providers.instancetype.capacity")
+	if !nodeutils.IsManaged(node, c.cloudProvider) {
```

Comment:
I'd rather not rely on global state, especially given this is used across packages. IMO the cloudprovider is appropriate since it's the cloudprovider implementation that dictates what's "managed" upstream.